engine/access/rest/apiproxy/rest_proxy_handler.go

```diff
@@ -0,0 +1,344 @@
+	r.log("upstream", "GetCollectionByID", err)
+
+	if err != nil {
+		return nil, err
+	}
+
+	transactions, err := convert.MessageToLightCollection(collectionResponse.Collection)
+	if err != nil {
+		return nil, err
+	}
+
+	return transactions, nil
+}
+
+// SendTransaction sends already created transaction.
+func (r *RestProxyHandler) SendTransaction(ctx context.Context, tx *flow.TransactionBody) error {
+	upstream, err := r.FaultTolerantClient()
+	if err != nil {
+		return err
+	}
+
+	transaction := convert.TransactionToMessage(*tx)
+	sendTransactionRequest := &accessproto.SendTransactionRequest{
+		Transaction: transaction,
+	}
+
+	_, err = upstream.SendTransaction(ctx, sendTransactionRequest)
+	r.log("upstream", "SendTransaction", err)
+
+	return err
+}
+
+// GetTransaction returns transaction by ID.
+func (r *RestProxyHandler) GetTransaction(ctx context.Context, id flow.Identifier) (*flow.TransactionBody, error) {
+	upstream, err := r.FaultTolerantClient()
+	if err != nil {
+		return nil, err
+	}
+
+	getTransactionRequest := &accessproto.GetTransactionRequest{
+		Id: id[:],
+	}
+	transactionResponse, err := upstream.GetTransaction(ctx, getTransactionRequest)
+	r.log("upstream", "GetTransaction", err)
+
+	if err != nil {
+		return nil, err
+	}
+
+	transactionBody, err := convert.MessageToTransaction(transactionResponse.Transaction, r.Chain)
+	if err != nil {
+		return nil, err
+	}
+
+	return &transactionBody, nil
+}
+
+// GetTransactionResult returns transaction result by the transaction ID.
+func (r *RestProxyHandler) GetTransactionResult(ctx context.Context, id flow.Identifier, blockID flow.Identifier, collectionID flow.Identifier) (*access.TransactionResult, error) {
+	upstream, err := r.FaultTolerantClient()
+	if err != nil {
+
+		return nil, err
+	}
+
+	getTransactionResultRequest := &accessproto.GetTransactionRequest{
+		Id:           id[:],
+		BlockId:      blockID[:],
+		CollectionId: collectionID[:],
+	}
+
+	transactionResultResponse, err := upstream.GetTransactionResult(ctx, getTransactionResultRequest)
+	r.log("upstream", "GetTransactionResult", err)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return access.MessageToTransactionResult(transactionResultResponse), nil
+}
+
+// GetAccountAtBlockHeight returns account by account address and block height.
+func (r *RestProxyHandler) GetAccountAtBlockHeight(ctx context.Context, address flow.Address, height uint64) (*flow.Account, error) {
+	upstream, err := r.FaultTolerantClient()
+	if err != nil {
+		return nil, err
+	}
+
+	getAccountAtBlockHeightRequest := &accessproto.GetAccountAtBlockHeightRequest{
+		Address:     address.Bytes(),
+		BlockHeight: height,
+	}
+
+	accountResponse, err := upstream.GetAccountAtBlockHeight(ctx, getAccountAtBlockHeightRequest)
+	r.log("upstream", "GetAccountAtBlockHeight", err)
+
+	if err != nil {
+		return nil, models.NewNotFoundError("not found account at block height", err)
```

Comment:
why `models.NewNotFoundError`?

Reply:
Same question, there could be multiple reasons why this call would return an error, we can't assume it's "account not found"